.github/actions/test-dotnet/action.yml

```
@@ -0,0 +1,59 @@
+﻿name: Run .NET tests
+description: >
+  Runs .NET unit and UI tests (with the Lombiq UI Testing Toolbox for Orchard Core:
+  https://github.com/Lombiq/UI-Testing-Toolbox), generates a test report and uploads UI testing failure dumps to
+  artifacts.
+
+inputs:
+  build-directory:
+    required: false
+    default: .
+    description: >
+      Path to the directory where a solution file can be found and thus the .NET build has run. Defaults to ".".
+  test-verbosity:
+    required: false
+    default: quiet
+    description: Verbosity parameter for dotnet test. Defaults to "quiet".
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Tests
+      shell: pwsh
+      working-directory: ${{ inputs.build-directory }}
+      run: |
+        ${{ github.action_path }}/run-tests.ps1 -Verbosity ${{ inputs.test-verbosity }}
+
+    # Note that uploading the failure dumps would fail under Windows if the path were too long, regardless of
+    # LongPathsEnabled, see: https://github.com/actions/upload-artifact/issues/309. To get around that we merge all
+    # FailureDumps into the solution root. The following two steps need to be success() || failure() (excluding the
+    # workflow being cancelled), so if there's a failure dump it'll be uploaded even if the tests are ultimately passed
+    # after retries.
+    # The same is true for the next step as well.
+    - name: Merge FailureDumps
+      shell: pwsh
+      if: (success() || failure()) && hashFiles(format('{0}/test/**/FailureDumps', inputs.build-directory))
+      run: |
+        ${{ github.action_path }}/merge-failure-dumps.ps1 -Directory "${{ inputs.build-directory }}"
+
+    - name: Upload UI testing artifacts
+      uses: actions/upload-artifact@v3
+      # We don't need additional conditions, because of the "if-no-files-found" setting.
+      if: success() || failure()
+      with:
+        name: ui-test-failure-dump-${{ inputs.build-directory }}-${{ matrix.machine-type }}
+        path: |
+          ${{ inputs.build-directory }}/FailureDumps/
+          test.out
+        if-no-files-found: ignore
+
+    - name: Test Report
+      # v8
+      uses: phoenix-actions/test-reporting@1e9b578818d03d6df50ce8e60d024a000bf3ab63
+      if: success() || failure()
+      with:
+        name: Test Results (${{inputs.build-directory}} ${{ matrix.machine-type }})
```

Now I think space is not a good separation because the build-directory may contain spaces, however unlikely...
```suggestion
        name: Test Results (${{inputs.build-directory}}|${{ matrix.machine-type }})
```